viam-cartographer_test.go

```diff
@@ -186,11 +223,183 @@ func TestNew(t *testing.T) {
+			Sensors:                 []string{"gibberish"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			DataRateMsec:            testDataRateMsec,
+			UseLiveData:             &_true,
+		}
+
+		_, err := internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeError,
+			errors.New("configuring lidar camera error: error getting lidar camera "+
+				"gibberish for slam service: \"rdk:component:camera/gibberish\" missing from dependencies"))
+	})
+
+	t.Run("Successful creation of cartographer slam service with good lidar when feature flag enabled", func(t *testing.T) {
+		termFunc := initTestCL(t, logger)
+		defer termFunc()
+
+		dataDirectory, fsCleanupFunc := initInternalState(t)
+		defer fsCleanupFunc()
+
+		attrCfg := &vcConfig.Config{
+			ModularizationV2Enabled: &_true,
+			Sensors:                 []string{"good_lidar"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			DataRateMsec:            testDataRateMsec,
+			UseLiveData:             &_true,
+		}
+
+		svc, err := internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, svc.Close(context.Background()), test.ShouldBeNil)
+	})
+
+	t.Run("Failed creation of cartographer slam service with invalid sensor "+
+		"that errors during call to NextPointCloud when feature flag enabled", func(t *testing.T) {
+		termFunc := initTestCL(t, logger)
+		defer termFunc()
+
+		dataDirectory, fsCleanupFunc := initInternalState(t)
+		defer fsCleanupFunc()
+
+		attrCfg := &vcConfig.Config{
+			ModularizationV2Enabled: &_true,
+			Sensors:                 []string{"invalid_sensor"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			DataRateMsec:            testDataRateMsec,
+			UseLiveData:             &_true,
+		}
+
+		_, err = internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeError,
+			errors.New("failed to get data from lidar: ValidateGetData timeout: invalid sensor"))
+	})
+
+	t.Run("Successful creation of cartographer slam service in localization mode when feature flag enabled", func(t *testing.T) {
+		termFunc := initTestCL(t, logger)
+		defer termFunc()
+
+		dataDirectory, fsCleanupFunc := initInternalState(t)
+		defer fsCleanupFunc()
+
+		attrCfg := &vcConfig.Config{
+			ModularizationV2Enabled: &_true,
+			Sensors:                 []string{"replay_sensor"},
+			ConfigParams:            map[string]string{"mode": "2d"},
+			DataDirectory:           dataDirectory,
+			UseLiveData:             &_true,
+			MapRateSec:              &_zeroInt,
+		}
+
+		svc, err := internaltesthelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeNil)
+
+		// TODO: Implement these
```

Comment:
These can't be implemented until the corresponding methods are supported when the feature flag is on.